docs/migration.md

```diff
@@ -1,235 +1,288 @@
+	`rdiff-backup-2`, or even z-Versions like `rdiff-backup-2.0.5` but
+	the middle version seems like a reasonable middle-way.
+
+Optionally, you can add to your PATH an executable script `rdiff-backup-2.0`
+with a content like the following, so that the clients don't need to care
+about the full-path (which will be our assumption in the following lines):
+
+```
+#!/bin/sh
+BASEDIR=/usr/local/lib
+exec ${BASEDIR}/$(basename $0)/bin/rdiff-backup "$@"
+```
+
+> **TIP:** the `basename` trick allows you to only maintain one script,
+	linked (hard or soft) under multiple names.
+
+Repeat for each version of rdiff-backup you want to maintain in parallel.
+
+
+### Client side
+
+The client side is even simpler, you just need to use **\--remote-schema**
+pointing at the correct version of rdiff-backup, something like:
+
+```
+rdiff-backup --remote-schema 'ssh -C %s rdiff-backup-2.0 --server' \
+	-b /sourcedir user@serverhost::/backup-repo
+```
+
+Starting with rdiff-backup 2.1+, the command would look like this and
+wouldn't need to be changed with each update of the client, as the version
+would automatically follow:
+
+```
+rdiff-backup --remote-schema 'ssh -C {h} rdiff-backup-{vx}.{vy} server' \
+	backup /sourcedir user@serverhost::/backup-repo
+```
+
+> **TIP:** for older versions of rdiff-backup, one could surely write a
+	wrapper script mimicking the same behaviour, using something along the
+	line of `$(rdiff-backup --version | awk -F'[. ]' '{print $2 "." $3}')`.
+
+And that's it for the side-by-side installation...
+
+## Migration from old to new CLI
+
+After version 2.0.5, a new Command Line Interface (CLI) has been introduced
+in rdiff-backup, which offers also a compatibility layer, which means that we
+have effectively three CLIs:
+
+1. the old CLI until 2.0.5
+2. the legacy CLI, mimicking the old one, starting with 2.1+ (and deprecated)
+3. the new CLI, starting with 2.1+ as well
+
+The following tables show the main differences between those three versions of
+the rdiff-backup CLI, using typical usage examples.
+
+> **NOTE:** the new features aren't explained, only the mapping from the old
+	syntax to the new one.
+
+The differences between the old and the legacy CLI are, obviously, limited and
+restricted to the restore use cases:
+
+|Description | old CLI | legacy CLI |
+|------------|---------|------------|
+| Restore an increment file | `rdiff-backup {backup-repo}/rdiff-backup-data/{dated-increment} {target-dir}` | `rdiff-backup --restore {backup-repo}/rdiff-backup-data/{dated-increment} {target-dir}` |
+
+The differences between the old and the new CLI are more important, especially
+because the new CLI as a more strict approach to the differentiation between
+actions and options.
+
+|Description | old CLI | new CLI |
+|------------|---------|---------|
+| backup | `rdiff-backup [-b] {source-dir} {target-dir}` (`-b` or `--backup-mode`) | `rdiff-backup backup {source-dir} {target-dir}` |
```

Comment:
I feel we also need to update the rdiff-backup.net website accordingly

Reply:
Agreed, but I would appreciate if someone else could do it.